### PR TITLE
Adjust / simplify Docker setup

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Migrate Docker setup from https://github.com/ElrondNetwork/rosetta
 - Reference auxiliary scripts from https://github.com/ElrondNetwork/rosetta-docker-scripts
 - Merge image definitions into a single image.
 - Use a custom entrypoint script for the Docker image.

### CODE REVIEW

File: devnet.env [hunk 1]
```diff
@@ -0,0 +1,13 @@
+OBSERVER_PUBKEY="PUBLIC_KEY_FROM_KEY_FILE"
+OBSERVED_SHARD=0
+GENESIS_BLOCK=b099dbc444824f5bc30bf194a071f43c8e2019ae6213a641f12974fa6a6db5fb
```
Comment (andreibancioiu) [inline devnet.env:3]: Chosen as `prevBlockHash` of https://devnet-gateway.elrond.com/block/0/by-nonce/1

File: mainnet.env [hunk 2]
```diff
@@ -0,0 +1,13 @@
+OBSERVER_PUBKEY="PUBLIC_KEY_FROM_KEY_FILE"
+OBSERVED_SHARD=0
+GENESIS_BLOCK=165821a3407bd0d5916f9710e203bad788053443430e724b7847e086b175d9ab
```
Comment (andreibancioiu) [inline mainnet.env:3]: Chosen as `prevBlockHash` of https://gateway.elrond.com/block/0/by-nonce/1

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
+FROM golang:1.17.6 as builder
+
+# Clone repositories
+WORKDIR /repos
+RUN git clone https://github.com/ElrondNetwork/rosetta-docker-scripts.git --branch=rc/v1 --depth=1
```
Comment (iulianpascalau) [inline Dockerfile:5]: we should add some TODOs to change these as soon as we merge the branches and create releases

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+RUN git clone https://github.com/ElrondNetwork/elrond-go.git --branch=rc/2022-july --depth=1
+RUN git clone https://github.com/ElrondNetwork/rosetta.git --branch=main --depth=1
+
+# Build rosetta
+WORKDIR /go/rosetta/cmd/rosetta
+RUN go build
+
+# Build node
+WORKDIR /go/elrond-go/cmd/node
+RUN go build -i -v -ldflags="-X main.appVersion=rc/2022-july"
```
Comment (iulianpascalau) [inline Dockerfile:18]: let's not hardcode the node's version but instead to take it from git, somehow

**Review by miiu96 — DISMISSED** [review]

File: Dockerfile (hunk 4 above)
Comment (andreibancioiu) [reply Dockerfile:18]: Fixed.

File: Dockerfile (hunk 3 above)
Comment (andreibancioiu) [reply Dockerfile:5]: Fixed.

**Review by iulianpascalau — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**